tachyon/zk/plonk/examples/fibonacci/fibonacci1_circuit.h

```diff
@@ -0,0 +1,173 @@
+#ifndef TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI1_CIRCUIT_H_
+#define TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI1_CIRCUIT_H_
+
+#include <array>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/zk/plonk/constraint_system/circuit.h"
+
+namespace tachyon::zk::plonk {
+
+template <typename F>
+struct AssignedRow {
+  AssignedCell<F> a;
+  AssignedCell<F> b;
+  AssignedCell<F> c;
+
+  AssignedRow() = default;
+  AssignedRow(AssignedCell<F>&& a, AssignedCell<F>&& b, AssignedCell<F>&& c)
+      : a(std::move(a)), b(std::move(b)), c(std::move(c)) {}
+};
+
+template <typename F>
+struct Fibonacci1Config {
+  using Field = F;
+
+  Fibonacci1Config(std::array<AdviceColumnKey, 3>&& advice, Selector selector,
```

Comment:
Since `std::array<>`  isn't able to be moved, please remove `&&`!

```suggestion
  Fibonacci1Config(std::array<AdviceColumnKey, 3> advice, Selector selector,
```